### PR TITLE
petsc petsc-complex: fix Linux build and test

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Follow up to #97438 for Linux build.

### CODE REVIEW

File: Formula/petsc.rb [hunk 1]
```diff
@@ -45,16 +45,15 @@ def install
     # Avoid references to Homebrew shims
     rm_f lib/"petsc/conf/configure-hash"
 
-    if OS.mac?
-      inreplace lib/"petsc/conf/petscvariables", Superenv.shims_path, ""
-    elsif File.readlines("#{lib}/petsc/conf/petscvariables").grep(Superenv.shims_path.to_s).any?
+    if OS.mac? || File.foreach("#{lib}/petsc/conf/petscvariables").any? { |l| l[Superenv.shims_path.to_s] }
```
Comment (cho-m) [inline Formula/petsc.rb:48]: Main differences:
- `grep(...)` doesn't seem to work with string input. Passing a regex like `grep(Regexp.new(Regexp.escape(Superenv.shims_path.to_s)))` seems to work, but ended up more lengthy
- `foreach` is more memory efficient that `readlines` or `read`, as it avoids reading entire file into memory.

File: Formula/petsc-complex.rb [hunk 2]
```diff
@@ -44,16 +44,15 @@ def install
     # Avoid references to Homebrew shims
     rm_f lib/"petsc/conf/configure-hash"
 
-    if OS.mac?
-      inreplace lib/"petsc/conf/petscvariables", Superenv.shims_path, ""
-    elsif File.readlines("#{lib}/petsc/conf/petscvariables").grep(Superenv.shims_path.to_s).any?
+    if OS.mac? || File.foreach("#{lib}/petsc/conf/petscvariables").any? { |l| l[Superenv.shims_path.to_s] }
       inreplace lib/"petsc/conf/petscvariables", Superenv.shims_path, ""
```
Comment (carlocab) [inline Formula/petsc-complex.rb:48]:
```suggestion
      inreplace lib/"petsc/conf/petscvariables", "#{Superenv.shims_path}/", ""
```
I think this is needed or else you leave paths like `/clang` which clearly won't work.

Comment (cho-m) [reply Formula/petsc-complex.rb:48]: Are we fine with existing bottles or do we want to rebuild/revision-bump?

Comment (carlocab) [reply Formula/petsc-complex.rb:48]: Could probably rebuild bottles. Not sure about `revision` bump. In the absence of complaints I'd suggest skipping it.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**